pci_info.py

```diff
@@ -0,0 +1,106 @@
+        orig_cfg = "io_%s_fvt" % pci['adapter_type']
+        new_cfg = "io_%s_%s_fvt" % (pci['adapter_type'], cfg_name)
+        if not os.path.exists("config/tests/host/%s.cfg" % orig_cfg):
+            logger.debug("ignoring pci address %s as there is no cfg for %s", pci['pci_root'], pci['adapter_type'])
+            continue
+        shutil.copy("config/tests/host/%s.cfg" % orig_cfg, "config/tests/host/%s.cfg" % new_cfg)
+
+        # adding info to input file
+        if not CONFIGFILE.has_section(orig_cfg) or not additional_params:
+            continue
+        input_params = CONFIGFILE.items(orig_cfg)
+        if not input_params:
+            continue
+        INPUTFILE.add_section(new_cfg)
+        for param in input_params:
+            try:
+                key = param[0]
+                if ':' not in param[1]:
+                    value = pci[param[1]]
+                else:
+                    index = param[1].split(':')[0]
+                    index_exact = param[1].split(':')[1]
+                    if index_exact == 'all':
+                        value = ",".join(pci[index])
+                    else:
+                        value = pci[index][int(index_exact)]
+                INPUTFILE.set(new_cfg, key, "\"%s\"" % value)
+            except:
+                pass
+        for param in additional_params:
+            key = param.split('=')[0]
+            value = param.split('=')[1]
+            INPUTFILE.set(new_cfg, key, "\"%s\"" % value)
+        output += "host_%s," % new_cfg
+    if output:
+        output = output[:-1]
+
+    # write to input file
+    if input_params:
+        with open(input_path, 'w+') as input:
+            INPUTFILE.write(input)
+        output += " --input-file %s" % input_path
+
+    # print avocado-setup command additional params
+    if output:
+        output = "--run-suite %s" % output
+        print output
+        
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--pci-address', dest='pci_addr',
+                        action='store', default='',
+                        help='pci address, comma separated')
+    parser.add_argument('--show-info', dest='show_info',
+                        action='store_true', default=False,
+                        help='Show the pci details')
+    parser.add_argument('--create-config', dest='create_cfg',
+                        action='store_true', default=False,
+                        help='Create test config and input files')
+    parser.add_argument('--additional-params', dest='add_params',
+                        action='store', default='',
+                        help='Additional parameters(key=value) to the input file, space separated')
+    args = parser.parse_args()
+    if args.pci_addr:
+        pci_details = pci.pci_info(args.pci_addr)
+    else:
+        pci_details = pci.all_pci_info()
+    if args.show_info:
+        pprint(pci_details)
```

Comment:
In general, consider using modules that are supported even in python3 , so that transition is smoother...

Reply:
I am not aware of any changes to pprint in python3. Any references please ?